### ISSUE TITLE
Message Banner has poor compatibility with assistive technology

### ISSUE THREAD

**Issue by adampage** [adampage/github-cli-test#12]
The HTML+CSS implementation of our Message Banner component has poor compatibility with assistive technology and is lacking in semantic quality. To improve it, we can make the following changes:

- Wrap the content in a suitable semantic element. `<aside>` or `<article>` may be a good fit.
- Mark up the title with an appropriately leveled `<h#>` heading element.
- Allow the icon to provide alternative text for cases where it is intended to be informative (e.g., Error).

## Applicable WCAG Success Criteria

- [`1.1.1` Non-text Content — Level A](https://www.w3.org/WAI/WCAG21/quickref/#non-text-content)
- [`1.3.1` Info and Relationships — Level A](https://www.w3.org/WAI/WCAG21/quickref/#info-and-relationships)
